platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/InMemoryResolutionResultBuilder.java

```diff
@@ -38,9 +40,15 @@
 public class InMemoryResolutionResultBuilder implements DependencyGraphVisitor {
 
     private final DefaultResolutionResultBuilder resolutionResultBuilder = new DefaultResolutionResultBuilder();
```

Comment:
I would say these are quite closely related. They both serve the same purpose, except one adds a serialization step in-between, while the other does not serialize and just holds the result in memory. 

The in-memory implementation directly calls methods on `DefaultResolutionResultBuilder` when visiting, but the streaming implementation serializes the data, and only calls methods on `DefaultResolutionResultBuilder` when de-serializing the data.

Reply:
Yes, but they read as _even more closely related_ than that because of the shared suffix.  I'm just thinking you should rename one of them so it reads clearer.

Reply:
Renamed `DefaultResolutionResultBuilder` to `ResolutionResultGraphBuilder`